src/api/app/models/user.rb

```diff
@@ -80,36 +78,20 @@ class User < ApplicationRecord
                       message:     'must be a valid email address.',
                       allow_blank: true }
 
-  # We want to validate the format of the password and only allow alphanumeric
-  # and some punctiation/base64 characters.
-  # The format must only be checked if the password has been set and the record
-  # has not been stored yet.
+  # only allow alphanumeric and some punctiation/base64 characters
   validates :password,
             format: { with:    %r{\A[\w\.\- /+=!?(){}|~*]+\z},
-                      message: 'must not contain invalid characters.',
-                      if:      Proc.new { |user| user.password_changed? && !user.password_was.nil? } }
+                      message: 'must not contain invalid characters.' },
+            allow_nil: true
 
   # We want the password to have between 6 and 64 characters.
   # The length must only be checked if the password has been set and the record
   # has not been stored yet.
   validates :password,
             length: { within:    6..64,
                       too_long:  'must have between 6 and 64 characters.',
-                      too_short: 'must have between 6 and 64 characters.',
-                      if:        Proc.new { |user| user.password_changed? && !user.password_was.nil? } }
-
-  validates :password_hash_type, inclusion: { in:      PASSWORD_HASH_TYPES,
-                                              message: "%{value} must be in the list of hash types." }
-
-  validate :password_changes_with_hash_type
-  # Checks that the password got updated after password hash type changed
-  def password_changes_with_hash_type
-    # rubocop:disable Style/GuardClause
-    if password_hash_type_changed? && !password_was.nil? && !password_changed?
-      errors.add(:password_hash_type, 'cannot be changed unless a new password has been provided.')
-    end
-    # rubocop:enable Style/GuardClause
-  end
+                      too_short: 'must have between 6 and 64 characters.' },
+            allow_nil: true
```

Comment:
I would even consider to drop this validation completely. has_secure_password already checks for less than or equal to 72 characters.

Reply:
Well, but it doesn't check for the minimum length. So I think we should keep it since it's covered with that validation.

Reply:
I suggest dropping password rules
https://blog.codinghorror.com/password-rules-are-bullshit/
However, you can consider this as a nitpick :wink: